project/src/main/java/com/google/sps/data/ServletsLibrary.java

```diff
@@ -0,0 +1,168 @@
+package com.google.sps.data;
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.vision.v1.*;
+
+import java.util.*;
+ 
+import java.io.IOException;
+
+public class ServletsLibrary{
+
+
+
+    public static ArrayList<Productset> listProductSets(String projectId, String computeRegion) throws IOException {
+
+    ArrayList<Productset> productsets = new ArrayList<>();
+
+    try (ProductSearchClient client = ProductSearchClient.create()) {
+        // A resource that represents Google Cloud Platform location.
+        String formattedParent = ProductSearchClient.formatLocationName(projectId, computeRegion);
+        // List all the product sets available in the region.
+        for (ProductSet productSet : client.listProductSets(formattedParent).iterateAll()) {
+        // Display the product set information
+        System.out.println(String.format("Product set name: %s", productSet.getName()));
+        System.out.println(
+            String.format(
+                "Product set id: %s",
+                productSet.getName().substring(productSet.getName().lastIndexOf('/') + 1)));
+        System.out.println(
+            String.format("Product set display name: %s", productSet.getDisplayName()));
+        System.out.println("Product set index time:");
+        System.out.println(String.format("\tseconds: %s", productSet.getIndexTime().getSeconds()));
+        System.out.println(String.format("\tnanos: %s", productSet.getIndexTime().getNanos()));
+
+        Productset productset = new Productset(productSet.getName().substring(productSet.getName().lastIndexOf('/') + 1),
+                                    productSet.getDisplayName());
+        // The set id of a product set is the last directory of a file path and to get this file path
+        // the index of the last backslash symbol is determined and a substring from the index to the end of the file path is made
+        // to get the set id of a product set.
+        productsets.add(productset);
+        }
+    }
+
+    return productsets;
+  }
+
+  public static void createProductSet(
+  String projectId, String computeRegion, String productSetId, String productSetDisplayName)
```

Comment:
Function arguments and the throws IOException after them should be either on the same line as the opening parenthesis or indented on the line(s) below it.